Framework/Algorithms/inc/MantidAlgorithms/ManualProgressReporter.h

```diff
@@ -0,0 +1,34 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
```

Comment:
Would it be possible to define this purely in the unit test? It then wouldn't need to be defined in the algorithms library at all and `FrameworkManager::Instance` wouldn't be needed either in the unit test.

Reply:
Would `ProgressNotificationEvents` be triggered if the algortihm is not in the `FrameworkManager`?

Reply:
The class could be registered manually by the unit test code in to the `AlgorithmManager` and then creating it with the `AlgorithmManager::create` would hook up the notifications.